File: Websitedangtintimkiemnhatro/Websitedangtintimkiemnhatro/Models/Motel.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Websitedangtintimkiemnhatro.Models
{
    public class Motel
    {
        [Key]
        public int Id { get; set; }
        public string IdNew { get; set; }
        public string Title { get; set; }
        public float Price { get; set; }
        public DateTime DateUpdate { get; set; }
        public DateTime DateDue { get; set; }
        public string Status { get; set; }
        public bool Verify { get; set; }
        public string Address { get; set; }
        public string Description { get; set; }
        public string Phone { get; set; }
        public string AreaZone { get; set; }
        public Detail Detail { get; set; }
        [ForeignKey("DetailId")]
        [Required]
        public int DetailId { get; set; }
        public City City { get; set; }
        [ForeignKey("CityId")]
        [Required]
        public int CityId { get; set; }
        public ICollection<Image> Images { get; set; }
        public Bill Bill { get; set; }
        public User User { get; set; }
        [ForeignKey("UserId")]
        [Required]
        public int UserId { get; set; }
    }
}

File: chu12112020/font12112020/src/app/model/Motel.ts
import { Detail } from '../model/Detail';

export class Motel {
    id: Int32Array;
    idNew: Int32Array;
    title: string;
    name: string;
    price: Float32Array;
    dateUpdate:Date;
    dateDue:Date;
    status:string;
    verify:boolean;
    address:string;
    description:string;
    phone:string;
    time:string;
    typemotel:string;
    areaZone:string;
    cityid: string;
    city: City;
    userid: Number;
    user: User;
    typeservice: string;
    images: Image[];
    bill: Bill;
    detail:Detail;
    provinceid: string;
    province: Province;
}

export class Image {
    imageMotel: string;
}

export class Province {
    id: string;
    name: string;
    cityid: string;
    city: City;
}

export class Bill {
    id: string;
    timeChoice: string;
    numberDatePublish: Int32Array;
    payMoney:Float32Array;
    motelId:Int32Array;
    sevicepriceId:Int32Array;
}

export class City {
    id: string;
    name: string;
}

export class User {
    id: Number;
    firstName: string;
    lastName: string;
    createdDate: Date;
    lastLogOnDate: Date;
    Email: string;
    gender: boolean;
    title: string;
    userImage:string;
}